site/docs/transform/impute.md

```diff
@@ -0,0 +1,123 @@
+---
+layout: docs
+title: Impute
+permalink: /docs/impute.html
+---
+
+To impute missing data in Vega-Lite, you can either use the `impute` transform, either via an [encoding field definition](#encoding) or via an [`transform`](#transform) array.
+
+The impute transform groups data and determines missing values of the `key` field within each group.  For each missing value in each group, the impute transform will produce a new tuple with the `impute`d field generated based on a specified imputation `method` (by using a constant `value` or by calculating statistics such as mean within each group).
+
+## Documentation Overview
+{:.no_toc}
+
+- TOC
+{:toc}
+
+{:#encoding}
+## Impute in Encoding Field Definition
+
+{: .suppress-error}
+```json
+// A Single View Specification
+{
+  "data": ... ,
+  "mark": ... ,
+  "encoding": {
+    "x": {
+      "field": ...,
+      "type": "quantitative",
+      "impute": {...},               // Impute
+      ...
+    },
+    "y": ...,
+    ...
+  },
+  ...
+}
+```
+
+An [encoding field definition](encoding.html#field-def) can include an `impute` definition object to generate new data objects in place of the missing data.
+
+The `impute` definition object can contain the following properties:
+
+{% include table.html props="frame,keyvals,method,value" source="ImputeParams" %}
+
+For `impute` in encoding, the grouping fields and the key field (for identifying missing values) are automatically determined. Values are automatically grouped by the specified fields of [mark property channels](encoding.html#mark-prop), [key channel](encoding.html#key) and [detail channel](encoding.html#detail). If x-field is `impute`d, y-field is the key field. Basically, any missing y-value in each group will lead to a new tuple imputed, and vice versa.
+
+In this example, we `impute` the `y` channel's field (`"y"`), so the `x`'s field (`"x"`) will be used as the `"key"`.
```

Comment:
having x-channel's field named `"x"` makes the text read a bit confusing. I think you should rename the field in all examples to be a and b, so x = a, y = b.  Thus, the text will make it clear that `x` is a channel, while `"a"` is a field name.